packages/@aws-cdk/aws-codedeploy/lib/pipeline-action.ts

```diff
@@ -0,0 +1,81 @@
+import actions = require('@aws-cdk/aws-codepipeline-api');
+import cdk = require('@aws-cdk/cdk');
+
+/**
+ * Construction properties of the {@link PipelineDeployAction CodeDeploy deploy CodePipeline Action}.
+ */
+export interface PipelineDeployActionProps extends actions.CommonActionProps {
+    /**
+     * The name of the CodeDeploy application to deploy to.
+     */
+    application: string;
+
+    /**
+     * The name of the CodeDeploy deployment group to deploy to.
+     */
+    deploymentGroup: string;
+
+    /**
+     * The source to use as input for deployment.
+     */
+    inputArtifact: actions.Artifact;
+}
+
+export class PipelineDeployAction extends actions.DeployAction {
+    constructor(parent: cdk.Construct, id: string, props: PipelineDeployActionProps) {
+        super(parent, id, {
+            stage: props.stage,
+            artifactBounds: { minInputs: 1, maxInputs: 1, minOutputs: 0, maxOutputs: 0 },
+            provider: 'CodeDeploy',
+            inputArtifact: props.inputArtifact,
+            configuration: {
+                ApplicationName: props.application,
+                DeploymentGroupName: props.deploymentGroup,
+            },
+        });
+
+        // permissions, based on:
+        // https://docs.aws.amazon.com/codedeploy/latest/userguide/auth-and-access-control-permissions-reference.html
+
+        const actionsOnApplicaiton = [
```

Comment:
Typo

Reply:
Also, is putting this in a variable first really more readable than listing them in `addActions` without the splat?

Reply:
I like using temporary variables to name expressions as well, as long as I can add more information by naming the expression than just writing it directly in the place where it would be used. To *me*, it would be obvious that the actions you add to this PolicyStatement would be "actions on application"... but I guess that might not be true for everyone.

In any case, do as you please, I don't care enough to make you change it.

And I also don't need to see the new revision, if you make the requested changes feel free to merge directly.

Reply:
I've inlined the local variables. Thanks for the review!